### ISSUE TITLE
Excessive portion of total test runtime is spent running poll-cv (42% or total time)

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#14332]
Tracking issue for reduction of total test runtime.

I did an analysis of all our upstream/master tests runs over last 72 hours (aggregating duration of individual test runs)
```
Row | total_duration_hours | max_duration_seconds | avg_duration_seconds | tests_ran | poll_strategy |  
1 | 2713.0 | 752.92 | 15.51 | 629801 | poll-cv |  
2 | 809.0 | 782.78 | 3.92 | 742342 | epollsig |  
3 | 806.0 | 824.75 | 3.91 | 742381 | epoll1 |  
4 | 802.0 | 817.69 | 3.89 | 742398 | epollex |  
5 | 758.0 | 441.63 | 3.56 | 766727 | poll |  
6 | 457.0 | 984.3 | 9.82 | 167377 | null |  
7 | 55.0 | 82.11 | 0.17 | 1181569 | none |  
```

Because poll-cv tests are slower than other pollers, it turns out we are spending 42% of time running poll-cv tests, which seems like we aren't using the resources efficiently.
If we reduce/optimize the poll-cv tests, we can reduce the total test duration significantly.

**Comment by jtattermusch** [comment]
@markdroth   @nicolasnoble  @dgquintas  @sreecha , how useful are the poll-cv tests? We can make an additional breakdown, but first I wanted to check that we really need those tests.

CC @matt-kwong

**Comment by dgquintas** [comment]
They are quite useful. Given how different poll-cv is from all the other pollers (it's much slower), it has surfaced many issues in the past.

**Comment by jtattermusch** [comment]
Ok, then we'll probably need to take a look into making them more efficient or do a bit less (e.g. fewer iterations in loops etc.) to compensate for them being so slow. If there are some obvious candidates for optimization, please let me know.

**Comment by jtattermusch** [comment]
Fixed by #18298.
